DSCResources/MSFT_xDnsServerForwarder/MSFT_xDnsServerForwarder.psm1

```diff
@@ -0,0 +1,64 @@
+function Get-TargetResource
+{
+    [OutputType([Hashtable])]
+    param
+    (
+        [Parameter(Mandatory)]
+        [ValidateSet('Yes')]
+        [string]$IsSingleInstance,
+        [string[]]$IPAddresses
+    )
+    Write-Verbose 'Getting current DNS forwarders.'
+    [array]$currentIPs = (Get-CimInstance -Namespace root\MicrosoftDNS -ClassName microsoftdns_server).Forwarders
+    $targetResource =  @{
+        IsSingleInstance = $IsSingleInstance
+        IPAddresses = @()
+    }
+    if ($currentIPs)
+    {
+        $targetResource.IPAddresses = $currentIPs
+    }
+    Write-Output $targetResource
+}
+
+function Set-TargetResource
+{
+    param
+    (
+        [Parameter(Mandatory)]
+        [ValidateSet('Yes')]
+        [string]$IsSingleInstance,
+        [string[]]$IPAddresses
+    )
+    if (!$IPAddresses)
+    {
+        $IPAddresses = @()
+    }
+    Write-Verbose 'Setting DNS forwarders.'
+    Set-CimInstance -Namespace root\MicrosoftDNS -Query 'select * from microsoftdns_server' -Property @{Forwarders = $IPAddresses}
```

Comment:
Any change you could break this line into multiple lines to make it 100 chars wide? E.g.
```powershell
Set-CimInstance `
-Namespace root\MicrosoftDNS `
-Query 'select * from microsoftdns_server' `
-Property @{Forwarders = $IPAddresses}
```